packages/eslint-plugin/typings/eslint-utils.d.ts

```diff
@@ -113,67 +113,67 @@ declare module 'eslint-utils' {
 
   export function isArrowToken(
     token: TSESTree.Token | TSESTree.Comment,
-  ): boolean;
+  ): token is TSESTree.PunctuatorToken & { value: '=>' };
```

Comment:
question:
did we want to make this generic on the value?

Reply:
i did that at begging but typescript was behaving in unexpected way and type-guard was not working as it should